Fix duplicate follow notifications for existing followers

obenland opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

Fixes #2438

**Important Note:** This PR changes what we consider a "successful" follow. Previously, any follow request that didn't error was considered successful. Now, follow requests from actors who are already following are processed but marked as unsuccessful (`$success = false`) to prevent duplicate notifications while still maintaining proper federation behavior.

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* Prevent duplicate email notifications when ActivityPub instances re-send Follow activities for actors that are already following
* Check if actor already follows before setting success flag in handle_follow()
* Follow requests from existing followers are now marked as `$success = false` (not successful) to prevent email notifications
* Maintain proper federation behavior - Accept activities are still sent for duplicate follows

### Other information:

- [x] Have you written new tests for your changes, if applicable?

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Set up a WordPress site with the ActivityPub plugin
* Have an ActivityPub account (e.g., from Mastodon) follow your WordPress blog
* Note that you receive a "New Follower" email notification
* Have the same account send another follow request (or wait for automatic retries from instances like Bonfire)
* Verify that you do NOT receive another email notification for the same follower
* Check that the follower still appears in your followers list
* Run the test suite with `npm run env-test` to verify the new test passes

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger a GitHub workflow that will create and push the entry into the branch. -->

<!-- Due to org permissions, the job may fail for PRs crated from a fork under GitHub organizations. -->
<!-- In this case, you can create entry manually with `composer changelog:add` and push it into the branch. -->

<!-- If no changelog entry is required for this PR, please add the "Skip Changelog" label. -->

-   [x] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Added - for new features
-   [ ] Changed - for changes in existing functionality
-   [ ] Deprecated - for soon-to-be removed features
-   [ ] Removed - for now removed features
-   [x] Fixed - for any bug fixes
-   [ ] Security - in case of vulnerabilities

#### Message

<!-- Add a changelog message here -->
Prevent duplicate email notifications for existing followers.

</details>

obenland committed:
Prevent duplicate follow notifications for existing followers

Fixes #2438

When an ActivityPub instance re-sends a Follow activity for an actor that's already following, the plugin now prevents sending duplicate email notifications. The follow request is still processed and an Accept is sent, but success is set to false to skip the notification.

Changes:
- Check if actor already follows before setting success in handle_follow()
- Add test to verify duplicate follows don't trigger notifications
- Maintain proper federation behavior while preventing email spam

obenland committed:
Use wp_list_pluck instead of array_map for better WordPress coding standards

obenland committed:
Remove redundant _delete_all_posts() call - already handled in tear_down()

obenland committed:
Add changelog for duplicate follow notification fix

pfefferle commented on includes/handler/class-follow.php:
```diff
@@ -40,13 +41,18 @@ public static function handle_follow( $activity, $user_ids ) {
-			$user_id,
-			$activity['actor']
-		);
+		$remote_actor = Followers::add( $user_id, $activity['actor'] );
```
With that check in place, we can skip the Follow process when `$already_following` is `true`!?

```suggestion
		if ( ! $already_following ) {
			$remote_actor = Followers::add( $user_id, $activity['actor'] );
		}
```

obenland replied:
Fixed in [f67c6f4](https://github.com/Automattic/wordpress-activitypub/pull/2452/commits/f67c6f49e49d4b253865e52d734b77bd147df09c).

obenland committed:
Refactor: Skip Followers::add() when actor already following

Simplified the follow handler logic based on PR feedback. When an actor
is already following a user, we now skip calling Followers::add() entirely
rather than calling it and marking the result as unsuccessful.

This makes the code more efficient and easier to understand while
maintaining the same behavior: duplicate follows don't trigger
notifications but still receive Accept responses.

obenland committed:
Merge branch 'trunk' into fix/duplicate-follow-notifications
